src/api/auth/auth.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Prisma, User } from '@prisma/client';
+import {
+  Injectable,
+  NotFoundException,
+  BadRequestException,
+  ConflictException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { PasswordService } from './password.service';
+import { Token } from './entities/token.entity';
+import { CreateUserInput } from './dto/create-user.input';
+
+import config from '../config';
+import { PrismaService } from 'nestjs-prisma';
+
+const { JWT_PRIVATE_KEY, JWT_EXPIRE_TIME } = config;
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly jwtService: JwtService,
+    private readonly prisma: PrismaService,
+    private readonly passwordService: PasswordService,
+  ) {}
+
+  async createUser(payload: CreateUserInput): Promise<User> {
+    const hashedPassword = this.passwordService.hashPassword(payload.password);
+
+    try {
+      return await this.prisma.user.create({
+        data: {
+          ...payload,
+          password: hashedPassword,
+        },
+      });
+    } catch (e: any) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError && e.code === 'P2002') {
```

Comment:
What is this code? Maybe we will need a constant for these kind of things

Reply:
I saw it in SO, is a code for a duplicate key constraint failure